pkg/bpf/bpffs.go

```diff
@@ -104,100 +118,275 @@ func OpenAfterMount(m *Map) error {
 	return nil
 }
 
+// removeMap removes a map from the BPF filesystem.
+func removeMap(name string) {
+	os.Remove(MapPath(name))
+}
+
+// RemoveAfterMount schedules a map to be deleted after the BPF filesystem has
+// been mounted. If the filesystem is already mounted, the map is deleted
+// immediately.
+func RemoveAfterMount(name string) {
+	mountMutex.Lock()
+	defer mountMutex.Unlock()
+
+	if mounted {
+		removeMap(name)
+	}
+
+	log.WithField(logfields.BPFMapName, name).Debug("bpffs is not mounted yet; adding to list of maps to remove once it is mounted")
+	delayedRemoves = append(delayedRemoves, name)
```

Comment:
What's is the `delayedRemoves`?

Reply:
Delayed removes is the list of deprecated BPF maps which are going to be removed after BPFFS is mounted. I made it specially for https://github.com/cilium/cilium/blob/master/pkg/maps/tunnel/tunnel.go#L62-L63

Reply:
TBH, I'm almost sure that the `os.Remove` call which is currently used in master is ineffective, because when `init` functions are executed when Cilium is started, BPFFS is not mounted yet. So it's de facto a small bugfix. ;)